openwisp_users/tests/test_api/test_api.py

```diff
@@ -0,0 +1,627 @@
+        with self.assertNumQueries(4):
+            r = self.client.get(path)
+        self.assertEqual(r.status_code, 404)
+
+    def test_organization_put_api(self):
+        org1 = self._get_org()
+        self.assertEqual(org1.name, 'test org')
+        self.assertEqual(org1.description, '')
+        path = reverse('users:organization_detail', args=(org1.pk,))
+        data = {
+            'name': 'test org change',
+            'is_active': False,
+            'slug': 'test-org-change',
+            'description': 'testing PUT',
+            'email': 'testorg@test.com',
+            'url': '',
+        }
+        with self.assertNumQueries(6):
+            r = self.client.put(path, data, content_type='application/json')
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(r.data['name'], 'test org change')
+        self.assertEqual(r.data['description'], 'testing PUT')
+
+    def test_organization_patch_api(self):
+        org1 = self._get_org()
+        self.assertEqual(org1.name, 'test org')
+        path = reverse('users:organization_detail', args=(org1.pk,))
+        data = {
+            'name': 'test org change',
+        }
+        with self.assertNumQueries(5):
+            r = self.client.patch(path, data, content_type='application/json')
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(r.data['name'], 'test org change')
+
+    def test_create_organization_owner_api(self):
+        user1 = self._create_user(username='user1', email='user1@email.com')
+        org1 = self._create_org(name='org1')
+        org1_user1 = self._create_org_user(user=user1, organization=org1)
+        path = reverse('users:organization_detail', args=(org1.pk,))
+        data = {'owner': {'organization_user': org1_user1.pk}}
+        with self.assertNumQueries(17):
+            r = self.client.patch(path, data, content_type='application/json')
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(r.data['owner']['organization_user'], org1_user1.pk)
+
+    def test_remove_organization_owner_api(self):
+        user1 = self._create_user(username='user1', email='user1@email.com')
+        org1 = self._create_org(name='org1')
+        org1_user1 = self._create_org_user(user=user1, organization=org1)
+        self._create_org_owner(organization_user=org1_user1, organization=org1)
+        path = reverse('users:organization_detail', args=(org1.pk,))
+        data = {'owner': {'organization_user': ''}}
+        with self.assertNumQueries(11):
+            r = self.client.patch(path, data, content_type='application/json')
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(r.data['owner'], None)
+
+    def test_organization_delete_api(self):
+        org1 = self._create_org(name='test org 2')
+        self.assertEqual(Organization.objects.count(), 2)
+        path = reverse('users:organization_detail', args=(org1.pk,))
+        r = self.client.delete(path)
```

Comment:
Hi, In some of the tests, I have not asserted the number of queries, because it was changing on different test commands, the tests were passing on some and failing on some.